pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -293,10 +314,16 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 		)
 		return configMetaErr
 	}
-
-	if state == clusterMembersChangedUnready || (cr.Spec.ConnectionsGenToProcess != cr.Status.LastConnectionGen) {
-		cr.Status.SpecGenerationToProcess = &cr.Generation
-		cr.Status.LastConnectionGen = cr.Spec.ConnectionsGenToProcess
+	newHash := calcConnectionsHash(&cr.Spec.Connections, cr.Namespace)
+	if state == clusterMembersChangedUnready || (newHash != cr.Status.LastConnectionHash) {
+		if cr.Status.SpecGenerationToProcess == nil {
+			cr.Status.SpecGenerationToProcess = &cr.Generation
```

Comment:
Curious about this part; what's the reason to initialize it to cr.Generation now?

Reply:
Hmm, does it matter? we can start by this being equal to Generation and go on from there.

Reply:
It doesn't matter a whole lot. It does make one curious why choose to start at that number rather than at e.g. 0 or 1.